text/0000-io.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: io
+- Start Date: 2016-07-09
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the addition of the io package to the standard library. This package will contain interfaces to address the common actions of reading and writing data. The ReadBuffer and WriteBuffer classes will also be moved from the net package into the new io package so that they are more easily discovered.
+
+# Motivation
+
+Most applications need to process some input and produce some output based on that input. Therefore, these operations should be raised above application specific packages so that they can be used in a generalized way.
+
+# Detailed design
+
+- The io package will contain the following interfaces:
+```pony
+interface Readable
+  """
+  Wraps the read method which returns up to len bytes from the input.
+  """
+  fun read(len: USize): Array[U8] iso^
+
+interface Writable
+  """
+  Wraps the write method which writes data to the output and returns false if the data was not written properly.
+  """
+  fun ref write(data: ByteSeq): Bool
+```
+
+- The net/ReadBuffer and net/WriteBuffer classes will be moved to the io package.
+
+- The following interfaces will be added to the encode package:
+```pony
+interface Decoder
+  """
+  Wraps the decode method which reads encoded data from the Reader and returns the decoded bytes.
+  """
+  fun decode(r: Reader): Array[U8] iso^
+
+interface Encoder
+  """
+  Wraps the encode method which writes the encoded data to any Writer
```

Comment:
I don't think it's accurate to say that an interface "wraps" the methods.  These docstrings should just say what the method does, instead of saying that they "wraps the method that does X".